### ISSUE TITLE
Worker node does not join the master after master node was created with 'disable-agent' parameter when recycle agent vm

### ISSUE THREAD

**Issue by ShylajaDevadiga** [rancher/k3s#853]
v0.9.1

**Describe the bug**
If the master node is created with 'disable-agent' option, and you decide to join the worker node using node_token, worker node does not get added to the master with Linode.

**To Reproduce**
Create Master node:
`curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--disable-agent" sh -`

Get the node token:
`cat /var/lib/rancher/k3s/server/node-token`

Join the worker node:
`curl -sfL https://get.k3s.io | K3S_URL=https://mymaster:6443 K3S_TOKEN=XXX sh -`

**Expected behavior**
Worker node should be listed while using:
`kubectl get nodes`

**Actual behavior**
`kubectl get nodes`
No resources found.

**Additional context**
On the master nodes the coredns and the helm-install pods are in Pending state.
`k get pods -A`
NAMESPACE     NAME                         READY   STATUS    RESTARTS   AGE
kube-system   helm-install-traefik-dzl5t   0/1     Pending   0          4m56s
kube-system   coredns-66f496764-5lcmn      0/1     Pending   0          4m55s

**Comment by davidnuzik** [comment]
This seems like it may happen when we "recycle" a vm (agent node) machine.

When we tested with a fresh VM (1 server and 1 agent) and get config right the first time everything seems fine.

If we had made a configuration error when joining an agent and try again I think is when we encounter the issue. Perhaps the agent did already join and we are trying to join again and this breaks things.

We need to determine:
- Is this an urgent issue to fix before GA release?
- Is this a non-important issue that is nice-to-have-fixed after GA release?

I would think that other people may run into this so there is some concern. Perhaps someone else may misconfigure and try to join an agent again only to see if never works.
